site-src/geps/gep-713.md

```diff
@@ -0,0 +1,496 @@
+* Establish a pattern for policy attachment which should be used for any
+  implementation specific policies used with Gateway API resources
+* Provide a way to distinguish between required and default values
+* Enable policy attachment at all relevant scopes, including Gateways, Routes,
+  Backends, and maybe Namespaces
+* Provide a means of attachment that works for both ingress and mesh uses of
+  this API
+* Provide a consistent way that both included and implementation-specific
+  policies can be interpreted
+
+## Out of scope
+
+* Define all potential policies that may be attached to resources
+* Design the full structure and configuration of policies
+
+## API
+
+This approach is building on concepts from all of the alternatives discussed
+below. This is very similar to the existing BackendPolicy resource in the API,
+but also borrows some concepts from the ServicePolicy proposal.
+
+### Policy Attachment for Ingress
+Attaching policy to Gateway resources for ingress use cases is relatively
+straightforward. A policy can reference the resource it wants to apply to.
+Access is granted with RBAC - anyone that has access to create a RetryPolicy in
+a given namespace can attach it to any resource within that namespace.
+
+![Simple Ingress Example](images/713-ingress-simple.png)
+
+To build on that example, it’s possible to attach policies to more resources.
+Each policy applies to the referenced resource and everything below it in terms
+of hierarchy. Although this example is likely more complex than many real world
+use cases, it helps demonstrate how policy attachment can work across
+namespaces. 
+
+![Complex Ingress Example](images/713-ingress-complex.png)
+
+### Policy Attachment for Mesh
+Although there is a great deal of overlap between ingress and mesh use cases,
+mesh enables more complex policy attachment scenarios. For example, you may want
+to apply policy to requests from a specific namespace to a backend in another
+namespace. 
+
+![Simple Mesh Example](images/713-mesh-simple.png)
+
+Policy attachment can be quite simple with mesh. Policy can be applied to any
+resource in any namespace but it can only apply to requests from the same
+namespace if the target is in a different namespace.
+
+At the other extreme, policy can be used to apply to requests from a specific
+workload to a backend in another namespace. A route can be used to intercept
+these requests and split them between different backends (foo-a and foo-b in
+this case).
+
+![Complex Mesh Example](images/713-mesh-complex.png)
+
+### Policy TargetRef API
+
+Each Policy resource should include a single `targetRef` field. It must not
+target more than one resource at a time, but it can be used target larger
+resources such as Gateways or Namespaces that may apply to multiple child
+resources. This field should have the following structure:
+
+```go
+// PolicyTargetReference identifies an API object to apply policy to.
+type PolicyTargetReference struct {
```

Comment:
Question from present me and potentially future us:
Why not label selectors here as well? I'm not advocating for them at all, rather the opposite. Will documenting them here help?
Take this or leave it, not attached to this comment at all.

Reply:
Yeah this is a good point. I added an explanation for the rationale to start with a single targetRef per policy instead of multiple targetRefs or a selector. I don't think it needs to be this way forever, but I'd rather start with a simpler API that can be expanded than introduce too much complexity up front.